drivers/sensor/bmp581/bmp581.c

```diff
@@ -0,0 +1,640 @@
+	struct bmp581_data *drv = (struct bmp581_data *)dev->data;
+	int ret = 0;
+	uint8_t odr_val = 0;
+
+	ret = reg_read(BMP5_REG_ODR_CONFIG, &odr_val, 1, dev);
+	if (ret != BMP5_OK) {
+		return ret;
+	}
+	odr_val = BMP5_SET_BITSLICE(odr_val, BMP5_ODR, odr->val1);
+	ret = reg_write(BMP5_REG_ODR_CONFIG, (const uint8_t *)&odr_val, 1, dev);
+	get_osr_odr_press_config(&drv->osr_odr_press_config, dev);
+
+	return ret;
+}
+
+static int soft_reset(const struct device *dev)
+{
+	int ret = 0;
+	const uint8_t reset_cmd = BMP5_SOFT_RESET_CMD;
+	uint8_t int_status = 0;
+
+	if (dev == NULL) {
+		return -EINVAL;
+	}
+
+	ret = reg_write(BMP5_REG_CMD, &reset_cmd, 1, dev);
+
+	if (ret == BMP5_OK) {
+		k_usleep(BMP5_DELAY_US_SOFT_RESET);
+		ret = get_interrupt_status(&int_status, dev);
+		if (ret == BMP5_OK) {
+			if (int_status & BMP5_INT_ASSERTED_POR_SOFTRESET_COMPLETE) {
+				ret = BMP5_OK;
+			} else {
+				ret = -EFAULT;
+			}
+		}
+	} else {
+		LOG_DBG("Failed perform soft-reset.");
+	}
+
+	return ret;
+}
+
+static int bmp581_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	if (dev == NULL) {
+		return -EINVAL;
+	}
+
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	struct bmp581_data *drv = (struct bmp581_data *)dev->data;
+	uint8_t data[6];
+	int ret = 0;
+
+	ret = reg_read(BMP5_REG_TEMP_DATA_XLSB, data, 6, dev);
+	if (ret == BMP5_OK) {
+		int32_t raw_temperature =
+			(int32_t)((uint32_t)(data[2] << 16) | (uint16_t)(data[1] << 8) | data[0]);
+		/*
+		 * Division by 2^16(whose equivalent value is 65536) is performed to get temperature
+		 * data in deg C
+		 */
+		sensor_value_from_double(&drv->last_sample.temperature, raw_temperature / 65536.0);
+
+		if (drv->osr_odr_press_config.press_en == BMP5_ENABLE) {
+			uint32_t raw_pressure = (uint32_t)((uint32_t)(data[5] << 16) |
+							   (uint16_t)(data[4] << 8) | data[3]);
+			/*
+			 * Division by 2^6(whose equivalent value is 64) is performed to get
+			 * pressure data in Pa
+			 */
+			sensor_value_from_double(&drv->last_sample.pressure, (raw_pressure / 64.0));
+		} else {
+			drv->last_sample.pressure.val1 = 0;
+			drv->last_sample.pressure.val2 = 0;
+		}
+	}
+
+	return ret;
+}
+
+static int bmp581_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	if (dev == NULL || val == NULL) {
+		return -EINVAL;
+	}
+
+	struct bmp581_data *drv = (struct bmp581_data *)dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ALTITUDE: {
+#if IS_ENABLED(CONFIG_BMP581_ALTITUDE_CALC_MATH)
+		/*
+		 *	Whitepaper regarding calculation of pressure altitude can be found in:
+		 *	https://www.weather.gov/media/epz/wxcalc/pressureAltitude.pdf
+		 *
+		 *	Following formula calculates the altitude in meters from pressure
+		 *      and temperature.
+		 */
+		double last_pressure = sensor_value_to_double(&drv->last_sample.pressure);
+		double last_temperature = sensor_value_to_double(&drv->last_sample.temperature);
+		double sealevel_pressure = 101325.0;
```

Comment:
I think having this altitude feature in the current form is missleading, to get good accurate altitude measurements from pressure, you need to feed in the actual pressure at sea level at the location you are at, typically this can be fetched from APIs on internet if you know your location. The pressure at sea level will depend alot on the weather and will change quite a bit just during a day. By hardcoding the sea-level pressure you will get inaccurate altitude, and this could be confusing for users. My suggestion is to remove this feature from the driver as it's missleading for users that are not aware of this. The only way I see this feature useful, is to calculate relative height if you compare samples taken close in time, but still I'm not sure if it should be in the driver and called altitude.

Reply:
I left it in because it was in @talhaHavadar's original PR. I'm totaly fine with removing the altitude calculation from the driver as altitude isn't a value you can read from the sensor.